crates/turborepo-lib/src/shim.rs

```diff
@@ -467,17 +459,19 @@ impl RepoState {
         let potential_turbo_roots = reference_dir
             .ancestors()
             .filter_map(|path| {
-                let has_package_json = path.join_component("package.json").exists();
+                let package_json = PackageJson::load(&path.join_component("package.json")).ok();
```

Comment:
This is the reason I pulled the `turbopath` thread in this PR. The load needed an `AbsoluteSystemPath` so rather than do it at this call site I plumbed it through as far back as I could justify.